templates/single-tapestry.php

```diff
@@ -9,6 +9,26 @@
  * @return object null
  */
 
+// str is evaluated to detect support of ES6
+// originally by user DaBs: https://gist.github.com/DaBs/89ccc2ffd1d435efdacff05248514f38
+echo "<script type='text/javascript'>
+    var str = 'class ಠ_ಠ extends Array {constructor(j = \"a\", ...c) {const q = (({u: e}) => {return { [`s${c}`]: Symbol(j) };})({});super(j, q, ...c);}}' + 
+    'new Promise((f) => {const a = function* (){return \"\u{20BB7}\".match(/./u)[0].length === 2 || true;};for (let vre of a()) {' +
+    'const [uw, as, he, re] = [new Set(), new WeakSet(), new Map(), new WeakMap()];break;}f(new Proxy({}, {get: (han, h) => h in han ? han[h] ' + 
+    ': \"42\".repeat(0o10)}));}).then(bi => new ಠ_ಠ(bi.rd));';
+
+    try {
+        eval(str);
+    } catch(e) {
+        var div = document.createElement(\"div\")
+        var message = document.createTextNode(\"To use the Tapestry Tool, please use a browser that supports ES6.\")
```

Comment:
This message is not very user friendly. Average Joe has no idea what ES6 is so we should say "Please update your browser" and list a list of links to common browser websites or instructions for updating their browsers. The messaging should be created for a very non-savvy user.

Reply:
This whole JS is better to be included as a separate JS file rather than being inserted inline.